indexing-hadoop/src/main/java/io/druid/indexer/hadoop/DatasourceInputSplit.java

```diff
@@ -0,0 +1,64 @@
+package io.druid.indexer.hadoop;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.mapreduce.InputSplit;
+
+public class DatasourceInputSplit extends InputSplit implements Writable
+{
+  //absolute hdfs path all the way to index.zip
+  private String path;
```

Comment:
as said in earlier comment, I'm thinking of keeping Map<String,Object> loadSpec here. that is consistent with DataSegment and has enough information to load the segment.